app/controllers/sample_transfers_controller.rb

```diff
@@ -32,6 +32,24 @@ def create
   def confirm
     transfer = SampleTransfer.find(params[:sample_transfer_id])
 
+    unless transfer.receiver_institution == @navigation_context.institution
```

Comment:
Sounds good. I was wondering whether there could be a more specific error than just "not found" to tell the user that they're trying to confirm from a context that is not the intended receiver. But that's probably not necessary. This route would only be hit when a specific link is generated and that should only work from the receiver context. We don't need to pay much attention to error handling. For that reason I don't think we need to do anything about the `RecordNotFound` exception. It just returns a 404 status and that's it.

There is a check missing to verify the transfer has not already been confirmed. I'm doing that now. This should result in a proper error message, because it's an error that is likely to occur in normal usage (changing the `confirmed` status may lead to race conditions).